com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs

```diff
@@ -184,10 +206,48 @@ public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderin
                 );
                 material.SetVector(s_SSAOParamsID, ssaoParams);
 
+#if ENABLE_VR && ENABLE_XR_MODULE
+                int eyeCount = renderingData.cameraData.xr.enabled && renderingData.cameraData.xr.singlePassEnabled ? 2 : 1;
+#else
+                int eyeCount = 1;
+#endif
+                for (int eyeIndex = 0; eyeIndex < eyeCount; eyeIndex++)
+                {
+                    Matrix4x4 view = renderingData.cameraData.GetViewMatrix(eyeIndex);
+                    Matrix4x4 proj = renderingData.cameraData.GetProjectionMatrix(eyeIndex);
+                    m_CameraViewProjections[eyeIndex] = proj * view;
```

Comment:
Why do we upload specific matrices here instead of using the camera matrices, `unity_CameraProjection` et al?        
https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/ShaderLibrary/UnityInput.hlsl#L28-L32

Reply:
We need to combined view proj matrix. Unity/XR team does not provide `unity_CameraViewProjection`, so I have to compute it manually. I talked to XR team about adding that but they don't think it is either high priority or really needed. I want to avoid calculating view proj matrix in shader code as the calculation runs per pixel, so I ended up having to write the extra scaffolding code to provide this matrix manually (as well as `unity_eyeIndex` mechanism, which works in both non XR and XR environment).